projects/lib/src/chessgame.cpp

```diff
@@ -267,7 +267,10 @@ void ChessGame::onMoveMade(const Chess::Move& move)
 
 	// Get the result before sending the move to the opponent
 	m_board->makeMove(move);
-	m_result = m_board->result();
+	const bool useRMobility = m_adjudicator.rMobilityEnabled();
```

Comment:
Like @alwey noticed, do we have to calculate r-mobility after every move? Couldn't we do it only if the game has ended? And maybe that calculation could be done in `m_board->result()` function?